tests/scenario/test_middlewares.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import unittest
+from unittest.mock import MagicMock, PropertyMock, patch
+
+from charm import TraefikIngressCharm
+from scenario import Container, Context, State, Relation
+
+
+@patch("charm.KubernetesServicePatch")
+@patch("lightkube.core.client.GenericSyncClient")
+class TestMiddlewares(unittest.TestCase):
+    def setUp(self) -> None:
+        self.containers = [Container(name="traefik", can_connect=True)]
+
+        version_patcher = patch.object(TraefikIngressCharm, "version", property(lambda *_: "0.0.0"))
+        self.version_patch = version_patcher.start()
+        self.addCleanup(version_patcher.stop)
+
+    @patch("charm.TraefikIngressCharm.external_host", PropertyMock(return_value="testhostname"))
+    @patch("charm.TraefikIngressCharm._traefik_service_running", PropertyMock(return_value=True))
+    @patch("charm.TraefikIngressCharm._tcp_entrypoints_changed", MagicMock(return_value=False))
+    def test_start_traefik_is_not_running(self, *_):
+        # GIVEN an ingress (per app) relation is requesting middlewares (strip prefix and
+        #  redirect-https)
+        relation = Relation(
+            endpoint="ingress",
+            interface="ingress",
+            remote_app_name="remote",
```

Comment:
this is the default

Reply:
The "remote" literal is all over the place so I'd rather have it explicit.